bin/annotate_test_failures

```diff
@@ -1,10 +1,27 @@
 #!/usr/bin/env ruby
 #
 # Usage:
-#   annotate_test_failures [junit-file-path]
+#   annotate_test_failures [junit-file-path] [should_send_slack_notification]
+#
+# Arguments:
+#   [junit-file-path]                Path to the JUnit XML file containing test results.
+#
+#   [should_send_slack_notification] Boolean flag indicating whether Slack notification should be sent.
+#                                    This argument is optional. If not provided, Slack notification will not be sent.
 #
 require 'rexml/document'
 require 'shellwords'
+require 'net/http'
+require 'json'
+
+BUILDKITE_ORGANIZATION_SLUG = ENV.fetch('BUILDKITE_ORGANIZATION_SLUG')
+BUILDKITE_PIPELINE_SLUG = ENV.fetch('BUILDKITE_PIPELINE_SLUG')
+BUILDKITE_BUILD_NUMBER = ENV.fetch('BUILDKITE_BUILD_NUMBER')
+BUILDKITE_BUILD_AUTHOR = ENV.fetch('BUILDKITE_BUILD_AUTHOR')
+BUILDKITE_PIPELINE_NAME = ENV.fetch('BUILDKITE_PIPELINE_NAME')
+BUILDKITE_JOB_ID = ENV.fetch('BUILDKITE_JOB_ID')
+BUILDKITE_BRANCH = ENV.fetch('BUILDKITE_BRANCH')
+SLACK_WEBHOOK = ENV.fetch('SLACK_WEBHOOK')
```

Comment:
`ENV.fetch(…)` will fail if the env var doesn't exist. Given that not all callers will want to use your new feature of sending messages to slack, they might not all have the `SLACK_WEBHOOK` env var declared in the first place if they don't really need it. But reading that value unconditionally at the start of the script will crash early anyway.

In theory you could use `ENV.fetch('SLACK_WEBHOOK', nil)` to provide a default value of `nil` instead of crashing when the env var is undefined. `ENV['SLACK_WEBHOOK']` syntax also works and also returns `nil` if undefined, so both syntaxes would be acceptable here.

In practice, I'd instead suggest to:
 - Not bother declaring those constants, and direclty use `ENV['BUILDKITE_ORGANIZATION_SLUG']` and the likes in your `send_slack_notification` method.
    - Defining those as constants doesn't feel a big win to me given all it wins you is saving 5 characters (`ENV[`+`]`)
    - It's ok to assume that all the `BUILDKITE_*` env vars are defined since this `annotate_test_failures` is made to only be called from Buildkite scripts (it doesn't even make sense to try creating a Buildkite annotation outside of the context of a Buildkite build anyway)
 - As for the `SLACK_WEBHOOK` one:
    - I also suggest you to read only from inside your `send_slack_notification` to not try to read it if the user didn't even intent to send a Slack notification in the first place
    - You could still consider using `ENV.fetch('SLACK_WEBHOOK')` syntax—and have it crash if the env var is undefined, because that ENV var would be required after all for sending slack notifications—but maybe it would be more user-friendly to catch those errors nicely (e.g. `webhook_url = ENV['SLACK_WEBHOOK']` then use something like `puts "Can't send the Slack notification because 'SLACK_WEBHOOK' env var is not defined." if webhook_url.nil?`
    - A third alternative would be to allow passing that webhook URL as a parameter to the script(†), so that regardless of the name of the ENV var that various pipelines and projects might be using to declare that URL (some projects might have it declared as `SLACK_BUILD_WEBHOOK` or `SLACK_URL` in their `.mobile-secrets` for example) the caller can decide what to use.

_(†) At that point we might consider using `OptionParser` to provide nicer-looking and more explicit parameters for the `annotate_test_failures` call, see comment below.

Reply:
yeah makes sense, updated the new env vars to this format `ENV['ENV_VAR_NAME']` as part of the same commit for the OptionParser change in https://github.com/Automattic/a8c-ci-toolkit-buildkite-plugin/pull/60/commits/351c7348152a3ddad0aa0c0d97ffc05a79b0734a